docs/userguide/storagedriver/aufs-driver.md

```diff
@@ -0,0 +1,194 @@
+
+Consider, for example, an application running in a container needs to add a single new value to large key-value store (file). If this is the first time the file is modified it does not yet exist in the container's top writable layer. So, the CoW must *copy up* the file from the underlying image. The AUFS storage driver searches each image layer for the file. The search order is from top to bottom. When it is found, the entire file is *copied up* to the container's top writable layer. From there, it can be opened and modified.
+
+Larger files obviously take longer to *copy up* than smaller files, and files that exist in lower image layers take longer to locate than those in higher layers. However, a *copy up* operation only occurs once per file on any given container. Subsequent reads and writes happen against the file's copy already *copied-up* to the container's top layer.
+
+
+## Deleting files with the AUFS storage driver
+
+The AUFS storage driver deletes a file from a container by placing a *whiteout
+file* in the container's top layer. The whiteout file effectively obscures the
+existence of the file in image's lower, read-only layers.  The simplified
+diagram below shows a container based on an image with three image layers.
+
+![](images/aufs_delete.jpg)
+
+The `file3` was deleted from the container. So, the AUFS storage driver  placed
+a whiteout file in the container's top layer. This whiteout file effectively
+"deletes"  `file3` from the container by obscuring any the original file's
+existence in the image's read-only base layer. Of course, the image could have
+been in any of the other layers instead or as well depending on how the layers
+are built.
+
+## Configure Docker with AUFS
+
+You can only use the AUFS storage driver on Linux systems with AUFS installed. Use the following command to determine if your system supports AUFS.
+
+```bash
+$ grep aufs /proc/filesystems
+nodev   aufs
+```
+
+This output indicates the system supports AUFS.  Once you've verified your
+system supports AUFS, you can must instruct the Docker daemon to use it. You do
+this from the command line with the `docker daemon` command:
+
+```bash
+$ sudo docker daemon --storage-driver=aufs &
+```
+
+Alternatively, you can edit the Docker config file and add the
+`--storage-driver=aufs` option to the `DOCKER_OPTS` line.
+
+```bash
+# Use DOCKER_OPTS to modify the daemon startup options.
+DOCKER_OPTS="--storage-driver=aufs"
+```
+
+Once your daemon is running, verify the storage driver with the `docker info` command.
+
+```bash
+$ sudo docker info
+Containers: 1
+Images: 4
+Storage Driver: aufs
+ Root Dir: /var/lib/docker/aufs
+ Backing Filesystem: extfs
+ Dirs: 6
+ Dirperm1 Supported: false
```

Comment:
Should we mention `Dirperm1`? This has been the biggest issue around AUFS (i.e. permissions not being applied correctly)